tutor/templates/jobs/init/cms.sh

```diff
@@ -16,3 +16,7 @@ fi
 # Create waffle switches to enable some features, if they have not been explicitly defined before
 # Copy-paste of units in Studio (highly requested new feature, but defaults to off in Quince)
 (./manage.py cms waffle_flag --list | grep contentstore.enable_copy_paste_units) || ./manage.py lms waffle_flag --create contentstore.enable_copy_paste_units --everyone
+
+# Re-index studio and courseware content
+./manage.py cms reindex_studio --experimental
+./manage.py cms reindex_course --active
```

Comment:
Should gate these too (though not sure about which variable to use to do this?)

```suggestion
{% if ACTIVATE_MEILISEARCH %}
./manage.py cms reindex_studio --experimental
./manage.py cms reindex_course --active
{% endif %}
```

Reply:
No, I don't think that's right. We assume that meilisearch is running, somewhere. Even if RUN_MEILISEARCH is false, it just means that meilisearch is running on a remote host. So we still need to run the migration commands.

Reply:
Oh yes, you're correct!

There's an issue with `reindex_course` referencing elasticsearch directly (https://github.com/openedx/edx-platform/pull/35743), but that's not Tutor's problem :)

Reply:
I suggest we run ` ./manage.py lms shell -c "import search.meilisearch; search.meilisearch.create_indexes()"` in init/lms.sh. What do you think?

Reply:
Also keep in mind that until we resolve https://github.com/openedx/modular-learning/issues/235 (which is in progress), we should avoid running `./manage.py cms reindex_studio --experimental` on any instances with thousands of courses or more.

Reply:
OK I'll comment out this command then.

Reply:
@bradenmacdonald Now that I think of it, it does seem overkill to re-index the studio but also courses in general for every call to `tutor local launch`. @DawoudSheraz also noticed that we re-indexed all courses during course import: https://github.com/overhangio/tutor/blob/e2786afa58106a9a006ee5aa224cc806ed1bb917/tutor/commands/jobs.py#L176

Is there any way we can be smarter about which courses to re-index, and when? For instance, it would be reasonable to automatically re-index a course after it has been created. 

In a perfect world, I would like to be able to get rid of all calls to `./manage.py cms reindex_studio` and `./manage.py cms reindex_course` in Tutor.  Do you think that's possible within a reasonable time frame?

Reply:
@regisb We only need to run `reindex_studio` on the initial launch of a new instance (because it creates the index and sets the required index parameters, much like a database migration), if the index configuration has changed (only affects tutor nightly / master branch of edx-platform), and presumably when the user upgrades to a new major version.

Running it on the initial launch will be extremely fast, assuming no courses/libraries exist.

So I would recommend putting it only in the upgrade / provisioning workflow and not in the general `launch` flow. Except maybe in development mode it should be in the `launch` as well, assuming people won't have too much data on their dev instances.

Reply:
As for re-indexing courses, I haven't looked into it but it does sound like we're doing it more than is necessary.

Reply:
Is there a function that would only create and configure the indexes (without deleting them first), such that we could call this function at every launch? Pretty much like a `./manage.py migrate` command.

Reply:
It doesn't exist yet, but we could add that. A `--create-only` flag would be trivial.

The edge case is when upgrading to a newer commit/release and the index parameters have changed; in that case, we need to replace the index and the new index will be empty unless we also re-index all the content. This could be done asynchronously though; it also doesn't replace the existing index until the reindexing is complete, so there's no downtime.

Reply:
I think we do need a lightweight, idempotent index creation command that works in just all cases. This can take different forms: a single python function, a management command, or a set of migrations. If we can get this in Sumac then that's great, if not then we should probably think about it for later releases.

Reply:
I added a proposal to https://github.com/openedx/modular-learning/issues/235 for a new `--init` flag that will achieve this.